sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/NamespacePropertiesExtensions.cs

```diff
@@ -3,34 +3,40 @@
 
 using System;
 using System.Globalization;
+using System.Threading.Tasks;
 using System.Xml.Linq;
+using Azure.Core.Pipeline;
 
 namespace Azure.Messaging.ServiceBus.Administration
 {
     internal class NamespacePropertiesExtensions
     {
-        public static NamespaceProperties ParseFromContent(string xml)
+        public static async Task<NamespaceProperties> ParseFromContentAsync(Response response, ClientDiagnostics diagnostics)
         {
             try
             {
+                string xml = await response.ReadAsStringAsync().ConfigureAwait(false);
                 var xDoc = XElement.Parse(xml);
                 if (!xDoc.IsEmpty)
                 {
                     if (xDoc.Name.LocalName == "entry")
                     {
-                        return ParseFromEntryElement(xDoc);
+                        return await ParseFromEntryElementAsync(xDoc, response, diagnostics).ConfigureAwait(false);
                     }
                 }
             }
             catch (Exception ex) when (!(ex is ServiceBusException))
             {
-                throw new ServiceBusException(false, ex.Message);
+                throw new ServiceBusException(false, ex.Message, innerException: ex);
             }
 
-            throw new ServiceBusException(false, "Unknown error.");
+            throw new ServiceBusException(
+                false,
+                "Unknown error.",
+                innerException: await diagnostics.CreateRequestFailedExceptionAsync(response).ConfigureAwait(false));
         }
 
-        private static NamespaceProperties ParseFromEntryElement(XElement xEntry)
+        private static async Task<NamespaceProperties> ParseFromEntryElementAsync(XElement xEntry, Response response, ClientDiagnostics diagnostics)
```

Comment:
I'm confused. The other 2 methods changed to just take response. Why not do the same here and put the xDoc boilerplate into the method? And the method name would be changed to reflect that. (same to all versions of this method from the extension classes)

Reply:
This one is acting on a specific XML element, but we need to pass along the response in case we throw.

Reply:
This is also used from the pageable API so that is it is extracted into a separate method.